server/src/utils.ts

```diff
@@ -6,7 +6,7 @@
  */
 
 export class Deferred<T> {
-    resolve: (value?: T) => void;
+    resolve: (value?: T | PromiseLike<T>) => void;
```

Comment:
In the updated Promise types the value is no longer optional so this should just be:
```suggestion
    resolve: (value: T) => void;
```